src/main/java/com/amazonaws/services/dynamodbv2/AmazonDynamoDBLockClient.java

```diff
@@ -497,14 +497,47 @@ public LockItem acquireLock(final AcquireLockOptions options) throws LockNotGran
                     item.put(RECORD_VERSION_NUMBER, AttributeValue.builder().s(String.valueOf(recordVersionNumber)).build());
                     sortKeyName.ifPresent(sortKeyName -> item.put(sortKeyName, AttributeValue.builder().s(sortKey.get()).build()));
                     newLockData.ifPresent(byteBuffer -> item.put(DATA, AttributeValue.builder().b(SdkBytes.fromByteBuffer(byteBuffer)).build()));
+                    Optional<AtomicLong> lastTouchedAt = options.getClockSkewUpperBound()
+                      .map(unused -> new AtomicLong(System.currentTimeMillis()));
+                    lastTouchedAt.ifPresent(value -> item.put(LAST_TOUCHED_AT, AttributeValue.builder().s(String.valueOf(value.get())).build()));
+
+                    if (options.shouldSkipBlockingWait() && existingLock.isPresent() && !existingLock.get().isExpired()) {
```

Comment:
This won't work and introduces a bug for backwards compatibility/those customers not using the new option. `existingLock.get().isExpired()` will always be false and this if statement will evaluate to `true` for people NOT using clockSkewUpperBound. Therefore, `LockCurrentlyUnavailableException` will be thrown whether the lock is expired or not.

Effectively this prevents people from acquiring a lock that existed before the deployment of this update.